<launch>
    <!--  Experiment arguments -->
    <arg name="experiment_name"             default="ExplorationPlanner"/>
    <arg name="experiment_config_embodied"  default="self_supervised/RunTurtlebot.yaml"/>     <!-- Non map related config (Model, Uncertainty, Online Learning, ... -->

    <arg name="planner_type"                default="mav_active_planner"/>  <!-- mav_active_planner | bump_and_rotate -->
    <arg name="planner_config"              default="exploration_planner_panoptic.yaml"/> <!-- Planner config with gain formulation -->
    <arg name="map_config"                  default="FullFlat/boundary.yaml"/> <!-- Map Boundaries to not drive in windows -->
    <!-- Mapper -->
    <arg name="config"                      default="single_tsdf"/> <!-- Map Configuration -->

    <!-- publish uncertainty -->
    <arg name="uncertainty"                 default="true"/>
    <arg name="node_start_delay"            default="10"/>
    <arg name="yaw_in_move_direction"       default="true"/>
    <arg name="delay"                       default="6.0"/> <!-- Experiment Startup Delay -->

    <!-- Airsim Node that Publishes Odometry and Images -->
    <arg name="sim_time"                    default="false"/>
    <arg name="mav_name"                    default="drone_1"/>
    <arg name="output"                      default="screen"/>
    <arg name="publish_clock"               default="true"/>
    <arg name="is_vulkan"                   default="true"/>
    <arg name="host"                        default="localhost"/>


    <!-- If Experiment should be resumed from previous state -->
    <arg name="resume"                      default = "false"/>
    <arg name="folder_to_resume"            default="/media/rene/Empty/presentation/Curiosity_planner_clustered_959_01__12_10_2021"/>
    <arg name="step_to_resume"              default="step_001" />
    <arg name="load_file"                   default="$(arg folder_to_resume)/map/$(arg step_to_resume)/current_map.panmap"/>


    <!-- Parse arguments to params -->
    <param name="folder_to_resume"      value="$(arg folder_to_resume)"/>
    <param name="step_to_resume"        value="$(arg step_to_resume)"/>
    <param name="map_to_load"           value="$(arg load_file)"/>
    <param name="resume"                value="$(arg resume)"/>
    <param name="experiment_name"       value="$(arg experiment_name)"/>

    <rosparam command="load"             file="$(find embodied_active_learning)cfg/experiments/environments/$(arg map_config)"/>
    <rosparam command="load"            file="$(find embodied_active_learning)cfg/experiments/$(arg experiment_config_embodied)"/>

    <group ns="experiment">
        <!-- Experiment launcher. Starts Planner, Captures Images etc. -->
        <node name="experiment_node" pkg="embodied_active_learning" type="turtlebot_experiment.py" output="screen"
              required="true" cwd="node">
            <param name="ns_planner" value="/planner/planner_node/planner"/>
            <param name="ns_airsim" value="/airsim/airsim_node"/>
            <param name="delay" value="$(arg delay)"/>
            <param name="startup_timeout" value="30.0"/>
            <remap from="rgbImage" to="/airsim/airsim_node/$(arg mav_name)/front/Scene"/>
            <remap from="depthImage" to="/airsim/airsim_node/$(arg mav_name)/front/DepthPlanner"/>
            <remap from="semsegImage" to="/airsim/airsim_node/$(arg mav_name)/front/Infrared"/>
        </node>
    </group>

    <node pkg="tf" type="static_transform_publisher" name="tf_drone_world" args="0 0 0 0 0 0 1 /drone_1 /world 100"/>
</launch>
